pkg/network/common/egressip.go

```diff
@@ -299,6 +366,108 @@ func (eit *EgressIPTracker) UpdateHostSubnetEgress(hs *osdnv1.HostSubnet) {
+	}
+}
+
+func (eit *EgressIPTracker) getNodeInternalIP(node *corev1.Node) string {
+	for _, nodeAddress := range node.Status.Addresses {
+		if nodeAddress.Type == corev1.NodeInternalIP && net.ParseIP(nodeAddress.Address).To4() != nil {
+			return nodeAddress.Address
+		}
+	}
+	return ""
+}
+
+// egressCIDROverlapsCloudSubnet checks that whatever egress CIDRs are specified
+// on the HostSubnet also are completely overlapped by the cloud network. A
+// failure to specify this correctly will lead to egress IP assignments by the
+// SDN which are not allowed by the cloud provider.
+func (eit *EgressIPTracker) egressCIDROverlapsCloudSubnet(hs *osdnv1.HostSubnet) error {
+	node, err := eit.nodeInformer.Lister().Get(hs.Host)
+	if err != nil {
+		return fmt.Errorf("error listing node, err: %v", err)
+	}
+	cloudEgressIPConfig, err := eit.getNodeCloudEgressIPConfig(node)
+	if err != nil {
+		return err
+	}
+	if cloudEgressIPConfig != nil && cloudEgressIPConfig.IFAddr.IPv4 != "" {
+		_, cloudNetwork, _ := net.ParseCIDR(cloudEgressIPConfig.IFAddr.IPv4)
+		for _, egressCIDR := range hs.EgressCIDRs {
+			_, egressSubnet, err := net.ParseCIDR(string(egressCIDR))
+			if err != nil {
+				return fmt.Errorf("invalid EgressCIDR: %v", egressCIDR)
+			}
+			if !containsCIDR(cloudNetwork, egressSubnet) {
+				return fmt.Errorf("EgressCIDR: %v does not fully overlap the cloud network: %v", egressSubnet, cloudNetwork)
+			}
+		}
+	}
+	return nil
+}
+
+func (eit *EgressIPTracker) getNodeCloudEgressIPConfig(node *corev1.Node) (*NodeCloudEgressIPConfiguration, error) {
+	nodeAnnotation := node.GetAnnotations()
+	nodeCloudEgressIPAnnotation, exists := nodeAnnotation[nodeEgressIPConfigAnnotationKey]
+	if !exists {
+		return nil, nil
+	}
+	cloudEgressIPConfig := []NodeCloudEgressIPConfiguration{}
+	if err := json.Unmarshal([]byte(nodeCloudEgressIPAnnotation), &cloudEgressIPConfig); err != nil {
+		return nil, fmt.Errorf("error de-serializing annotation: %q, err: %v", nodeCloudEgressIPAnnotation, err)
+	}
+	if len(cloudEgressIPConfig) > 0 {
+		return &cloudEgressIPConfig[0], nil
+	}
+	return nil, nil
+}
+
+func (eit *EgressIPTracker) initNodeCapacity(node *corev1.Node) error {
+	nodeIP := eit.getNodeInternalIP(node)
+	if nodeIP == "" {
+		return fmt.Errorf("node does not have an IPv4 InternalIP address")
+	}
+	if nodeEgress, exists := eit.nodesByNodeIP[nodeIP]; exists && nodeEgress.capacity == unlimitedNodeCapacity {
+		cloudEgressIPConfig, err := eit.getNodeCloudEgressIPConfig(node)
+		if err != nil {
+			return err
+		}
+		if cloudEgressIPConfig == nil {
+			return nil
+		}
+		// IP and IPv4 are mutually exclusive, so either one or the other is
+		// guaranteed to be set if this annotation exists.
+		if cloudEgressIPConfig.Capacity.IP != 0 {
+			nodeEgress.capacity = cloudEgressIPConfig.Capacity.IP
+		} else {
+			nodeEgress.capacity = cloudEgressIPConfig.Capacity.IPv4
+		}
+		klog.Infof("Initialized capacity: %v for Node: %q", nodeEgress.capacity, node.Name)
```

Comment:
Not sure I follow, do you mean something along the lines of: `klog.Infof("Initialized egress IP capacity: %v for Node: %q"...` ? It will log the message as coming from `egressip.go`, so it should be clear that this is a "egress IP related capacity" when reading the logs.

Reply:
yes, log messages should generally be obvious even without looking at the filename

Reply:
OK, I will update that now.